src/Categories/Adjoint/Parametric.agda

```diff
@@ -0,0 +1,150 @@
+
+      hom(L(c , x) , y) ----------------→ hom(x , R(c , y))
+        ↑                                          ↑
+        |                                          |
+        |                                          |
+        |                                          |
+      hom(L(c' , x) , y) ---------------→ hom(x , R(c' , y))
+
+    is commutative; unfolding the definitions, it seems that proving this exactly amounts to show
+    that the *unit* η c of the adjunction is a *wedge*. Viceversa, one also needs this naturality
+    when proving that the cowedge condition holds, so they seem to be logically equivalent.
+    Mac Lane (IV.7.Thm 3) seems to claim, however, that this condition needs not to be assumed
+    and can be derived from a better definition of parametric adjunction that we are not
+    currently able to devise.
+
+    More info at: https://math.stackexchange.com/questions/4581092/parametric-adjunctions-induce-cowedges-a-real-proof
+  -}
+
+  field
+    param-nat : ∀ {c c' x y} {f : c C.⇒ c'} {u : L₀.₀ c' x E.⇒ y} →
+      R₀.₁ c u D.∘ R₀.₁ c (η (L.₁ f) x) D.∘ η (A.unit c) x D.≈
+      η (R.₁ f) y D.∘ R₀.₁ c' u D.∘ η (A.unit c') x
+
+  -- Every parametric adjunction Lc ⊣ Rc defines a bifunctor L(-,R(-,x)) : Cᵒ x C → E
+  PABifunctor : {X : E.Obj} → Bifunctor C.op C E
+  PABifunctor {X} = record
+    { F₀ = λ { (c , c') → L₀.₀ c' (R₀.F₀ c X)}
+    ; F₁ = λ { {a , a'} {b , b'} (f , f') →
+        η (L.₁ f') (R₀.F₀ b X) E.∘ L₀.₁ a' (η (R.₁ f) X)}
+    ; identity = λ { {a , a'} →
+        begin _ ≈⟨ L.identity ⟩∘⟨ Functor.F-resp-≈ (L.F₀ a') R.identity ⟩
+              _ ≈⟨ refl⟩∘⟨ Functor.identity (L.F₀ a') ⟩
+              _ ≈⟨ E.identity² ⟩
+              _ ∎ }
+    ; homomorphism = λ { {a , a'} {b , b'} →
+        begin _ ≈⟨ L.homomorphism ⟩∘⟨ Functor.F-resp-≈ (L.F₀ a') R.homomorphism ⟩
+              _ ≈⟨ (refl⟩∘⟨ Functor.homomorphism (L.F₀ a')) ⟩
+              _ ≈⟨ E.assoc ⟩
+              _ ≈⟨ (refl⟩∘⟨ MR.pullˡ E (NaturalTransformation.commute (L.₁ _) _)) ⟩
+              _ ≈⟨ MR.assoc²'' E ⟩
+              _ ∎ }
+    ; F-resp-≈ = λ { {a , a'} {b , b'} {f} {g} (f≈g , f'≈g') →
+        L.F-resp-≈ f'≈g' ⟩∘⟨ Functor.F-resp-≈ (L.F₀ a') (R.F-resp-≈ f≈g)}
+    } where
+        open E.HomReasoning
+        open MorphismReasoning E
+
+  -- this is the main theorem of the module: a parametric adjunction Lc ⊣ Rc has a
+  -- counit ε[c,x] : L(c,R(c,x)) → x; this map is a cowedge in c.
+  counitCowedge : ∀ {A : Category.Obj E} → Cowedge {C = C} {D = E} (PABifunctor {A})
+  counitCowedge {A} = record
+    { E = A
+    ; dinatural = dtHelper record
+      { α = λ c → A.counit.η c _
+      ; commute = λ {X} {Y} f →
+        let
+          open Adjoint
+          adjunction-isoˡ : A.Ladjunct X (A.counit.η X A E.∘ L₀.₁ X (η (R.₁ f) A)) D.≈
+                            η (R.₁ f) A
+          adjunction-isoˡ = let open D.HomReasoning in
+            begin _ ≈⟨ homomorphism (R.F₀ X) ⟩∘⟨refl ⟩
+                  _ ≈⟨ D.assoc ⟩
```

Comment:
The various `pull` and `push` combinators can help shorten these proofs (here by 2 lines).